src/data/resolvers/user.ts

```diff
@@ -155,5 +163,45 @@ export const userResolvers = ({
         }),
       );
     },
+    async processedColonies({ colonyAddresses }) {
+      try {
+        const userColonies: Array<{
+          colonyChainId: string;
+          ensName: string;
+          metadata: string;
+          id: string;
+        }> = [];
+        const { data } = await apolloClient.query<
+          SubgraphColoniesQuery,
+          SubgraphColoniesQueryVariables
+        >({
+          query: SubgraphColoniesDocument,
+          fetchPolicy: 'network-only',
+        });
+        if (data?.colonies) {
+          colonyAddresses.map((colonyAddress) => {
+            const subscribedColony = ((data?.colonies as unknown) as Array<{
+              id: string;
+              colonyChainId: string;
+              ensName: string;
+              metadata: string;
+            }>).find(({ id }) => id === colonyAddress.toLowerCase());
+            if (subscribedColony) {
+              userColonies.push(subscribedColony);
+            }
+            return null;
```

Comment:
I don't understand how that would be more beneficial here ?

Sure it's a different approach to this sure, and it would remove the need for returns, but I don't see the benefit of using this over my current implementation.

Reply:
So "map" is used to "decorate" values in the array, not to loop over array and create different array

Reply:
I mean, I've been using `map` to modify an array my whole life :)

But I guess what you're saying makes sense, so I'll make the change.